lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -737,6 +738,9 @@ defmodule ExUnit.Case do
     tags
   end
 
+  defp validate_name(name) when byte_size(name) <= 255, do: name
+  defp validate_name(_name), do: raise("test name too long, it must be up to 255 characters")
```

Comment:
Let's refactor this to have a more actionable error message as well as to return the description as an atom itself:

```suggestion
  defp validate_test_name(name) do
    if byte_size(name) <= 255 do
      String.to_atom(name)
    else
      raise """
      the computed name of a test (which includes its type, the name of its parent describe \
      block if present, and the test name itself) must be shorter than 255 characters, \
      got: #{inspect(name)}
      """
  end
```